### PR TITLE
Switch plugin documentation to RDOC and mark advanced parameters

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ehelms** [comment]
Friendly ping for any maintainers since its been a few weeks without any review.

**Review by ekohl — COMMENTED** [review]
I don't see the benefit in this except for the cases where there are advanced parameters. puppet-strings is the official way of documenting and RDoc is one that only we use via Kafo. Ideally we'd find a way to get rid of RDoc completely and standardize on puppet-strings (to follow the ecosystem) but I'm not sure how to do that at this point. That's why I've chosen to use puppet-strings everywhere except when there are advanced parameters.

**Comment by ares** [comment]
Hold on, why are we closing this? Does this do any harm for outside of foreman-installer use case? I totally see the value for foreman-installer. Unless we're close to correctly parse this from puppet-strings, why do we block foreman-installer improvements (supposing the answer for the first question is no).

**Comment by ekohl** [comment]
I was arguing that we should prefer puppet-strings where it doesn't make a difference and RDoc where it does. Then it was closed.

**Comment by ehelms** [comment]
This was discussed partially out of band of the PR. The disconnect between different document styles makes choosing an already disjoint path forward harder. Given I cannot see how to reconcile that systematically at the moment, I left things as is.

File: manifests/plugin/memcache.pp [hunk 1]
```diff
@@ -6,6 +6,8 @@
 #
 # $hosts::      an array of hosts running memcache
 #
+# === Advanced parameters:
```
Comment (ekohl) [inline manifests/plugin/memcache.pp:9]: This one still makes sense

**Pull request closed without merging**